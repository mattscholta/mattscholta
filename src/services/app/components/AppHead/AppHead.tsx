import * as React from 'react';
import Head from 'next/head';
import { useRouter } from 'next/router';

import { SITE_TITLE } from '@src/config/constants';
import {
  BASE_URL,
  GOOGLE_ANALYTICS,
  GOOGLE_TAG_MANAGER
} from '@src/config/settings';
import { metadata } from '@src/services/seo/metadata';
import { GoogleAnalytics } from '@src/services/tracking/components/GoogleAnalytics';
import { GoogleTagManagerScript } from '@src/services/tracking/components/GoogleTagManagerScript';

interface AppHeadProps {}

/**
 * @name AppHead
 * @description A simple component that we use to set the "base" information
 * used in our Document <head>. This can be further "extended" by any
 * sub-routes, this is simply a strong base
 */
const AppHead: React.FC<AppHeadProps> = (_props) => {
  // Hooks
  const { pathname } = useRouter();

  // Setup
  const canonical = `${BASE_URL}${pathname}`;

  // Markup
  const renderMetaTag = (
    props: React.MetaHTMLAttributes<HTMLMetaElement>,
    index: number
  ) => <meta {...props} key={index} />;

  return (
    <Head>
      <GoogleAnalytics id={GOOGLE_ANALYTICS} />
      <GoogleTagManagerScript id={GOOGLE_TAG_MANAGER} />
      <link href={canonical} rel="canonical" />
      <link href="/favicon.ico" rel="icon" />
      <link href="/images/icon.png" rel="apple-touch-icon" sizes="48x48" />
      <link href="/manifest.json" rel="manifest" />
      {metadata.map(renderMetaTag)}
      <title>{SITE_TITLE}</title>
    </Head>
  );
};

export { AppHead };
export type { AppHeadProps };
