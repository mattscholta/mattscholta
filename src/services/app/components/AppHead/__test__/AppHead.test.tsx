import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { AppHead, AppHeadProps } from '../AppHead';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      asPath: '',
      pathname: '',
      push: jest.fn(),
      query: {},
      route: '/'
    };
  }
}));

describe('AppHead', () => {
  let props: AppHeadProps;
  let wrapper: RenderResult;

  const createWrapper = () => <AppHead {...props} />;

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
