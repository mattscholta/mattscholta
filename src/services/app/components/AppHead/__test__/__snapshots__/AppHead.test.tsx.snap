// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHead matches snapshot 1`] = `NodeList []`;
